diff --git a/src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java b/src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java
index 938f20f..d028b0d 100644
--- a/src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java
+++ b/src/main/java/dev/j3fftw/litexpansion/machine/AdvancedSolarPanel.java
@@ -25,7 +25,7 @@ import org.bukkit.World;
 import org.bukkit.block.Block;
 import org.bukkit.inventory.ItemStack;
 
-public class AdvancedSolarPanel extends SlimefunItem implements InventoryBlock, EnergyNetProvider {
+public class AdvancedSolarPanel extends SlimefunItem implements EnergyNetProvider {
 
     private static final int PROGRESS_SLOT = 4;
     private static final CustomItem generatingItem = new CustomItem(Material.ORANGE_STAINED_GLASS_PANE,
@@ -49,7 +49,7 @@ public class AdvancedSolarPanel extends SlimefunItem implements InventoryBlock,
         super(Items.LITEXPANSION, type.getItem(), RecipeType.ENHANCED_CRAFTING_TABLE, type.getRecipe());
         this.type = type;
 
-        createPreset(this, type.getItem().getImmutableMeta().getDisplayName().orElse("&7Solar Panel"),
+        new BlockMenuPreset(this.getId(), type.getItem().getImmutableMeta().getDisplayName().orElse("&7Solar Panel"),
             blockMenuPreset -> {
                 for (int i = 0; i < 9; i++)
                     blockMenuPreset.addItem(i, ChestMenuUtils.getBackground(), ChestMenuUtils.getEmptyClickHandler());
@@ -130,16 +130,6 @@ public class AdvancedSolarPanel extends SlimefunItem implements InventoryBlock,
         return this.type.getStorage();
     }
 
-    @Override
-    public int[] getInputSlots() {
-        return new int[0];
-    }
-
-    @Override
-    public int[] getOutputSlots() {
-        return new int[0];
-    }
-
     @Getter
     @AllArgsConstructor(access = AccessLevel.PRIVATE)
     public enum Type {
